engine/src/flutter/lib/ui/painting.dart

```diff
@@ -5517,6 +5546,19 @@ base class FragmentShader extends Shader {
     _floats = _constructor(_program, _program._uniformFloatCount, _program._samplerCount);
   }
 
+  List<UniformFloatSlot> _getSlotsForUniform(String name, int size) {
+    final int dataSize = _program._getUniformVectorSize(name);
+    assert(dataSize == size, 'Uniform `$name` has size $dataSize, not size $size.');
```

Comment:
Why even have the argument to this function if we can derive it correctly anyways?  I'd just check the size of the returned list and throw an error if they don't match.  We don't want to hit asserts in response to bad user interactions.  Asserts should be reserved for developer errors.